docs/modules/integrate/pages/mongodb-connector.adoc

```diff
@@ -1,3 +1,129 @@
+                 "mongodb://127.0.0.1:27017",
+                 "myDatabase",
+                 "myCollection"
+         );
+
+Pipeline p = Pipeline.create();
+(...)
+someStage.writeTo(mongoSink);
+```
+
+You can also use sink builder syntax:
+```java
+Sink<Document> mongoSink =
+         MongoSinks.builder(
+                     Document.class, <1>
+                     () -> MongoClients.create("mongodb://127.0.0.1:27017") <2>
+                 )
+                 .into("myDatabase", "myCollection") <3>
+                 .identifyDocumentBy("_id", doc -> doc.get("_id")) <4>
+                 .build()
+         );
+
+Pipeline p = Pipeline.create();
+(...)
+someStage.writeTo(mongoSink);
+```
+
+<1> Class of the document, that will be inserted. Can be `Document`, `BsonDocument` or any user defined POJO.
+<2> Supplier that creates mongo client or a `dataConnectionRef`
+<3> Destination database and collection to which items will be saved. In this case it's a constant value. You can also use overloaded method, that have 2 functions as arguments to specify target database and collection for every item separately (so in theory every item can land in different collection).
+<4> Specification of how the documents will be identified by the sink. This is required to distinguish two documents and know if we should perform insert or update. By default, if the class of the inserted document is `org.bson.Document`, the connector will distinguish two documents by their `_id` fields.
+
+Some other builder methods worth mentioning:
+
+1. `commitRetryStrategy` - how often a commit should be retried in case of transient errors. Used only with exactly once processing guarantee. Note that commit interval depends on the snapshot interval (more on this in <<Fault Tolerance>> section below), so commit interval in the Retry Strategy is a "minimum time", not exact value.
+2. `transactionOptions` - specifies Mongo's transaction options - read concern, write concern, read preference, etc.  Used only with exactly once processing guarantee.
+3. `writeMode` - `INSERT_ONLY`, `UPDATE_ONLY`, `UPSERT` or `REPLACE` - which operation will be used to put documents into Mongo. Default value is `REPLACE` (but inserts will be still performed if field got from `identifyDocumentBy` returns null).
+4. `throwOnNonExisting` - if true, connector will throw an exception if the database or collection does not exist prior to job execution.
+5. `withCustomReplaceOptions` - allows user to customize replace operations, like adding validation bypass or disabling upserts.
+
+By default sink puts the documents in parallel on all nodes.
+
+=== Fault Tolerance
+
+The Mongo sink supports exactly-once guarantee. It uses MongoDB's transactions if such guarantee is requested by the user. In case of transient errors, the transaction will be automatically retried as configured by `commitRetryStrategy` option. The documents are committed with the last snapshot; have in mind, that this increases latency.
```

Comment:
```suggestion
The Mongo sink supports exactly-once guarantee. It uses MongoDB's transactions if such guarantee is requested by the user. In case of transient errors, the transaction will be automatically retried as configured by the `commitRetryStrategy` option. The documents are committed with the last snapshot, which increases latency.
```